cpp/src/reductions/reductions.cpp

```diff
@@ -177,15 +177,16 @@ std::unique_ptr<scalar> reduce(column_view const& col,
         std::move(*reduction::detail::make_empty_histogram_like(col.child(0))), true, stream, mr);
     }
 
-    if (output_dtype.id() == type_id::LIST) {
-      if (col.type() == output_dtype) { return make_empty_scalar_like(col, stream, mr); }
-      // Under some circumstance, the output type will become the List of input type,
-      // such as: collect_list or collect_set. So, we have to handcraft the default scalar.
+    if (agg.kind == aggregation::COLLECT_LIST || agg.kind == aggregation::COLLECT_SET) {
       auto scalar = make_list_scalar(empty_like(col)->view(), stream, mr);
       scalar->set_valid_async(false, stream);
       return scalar;
     }
-    if (output_dtype.id() == type_id::STRUCT) { return make_empty_scalar_like(col, stream, mr); }
+
+    // `make_default_constructed_scalar` does not support nested type.
+    if (output_dtype.id() == type_id::STRUCT || output_dtype.id() == type_id::LIST) {
```

Comment:
Can we use `is_nested` here? https://github.com/rapidsai/cudf/blob/efae666bac226dc50c1c7b5d7f1145ee9a31fc66/cpp/src/utilities/traits.cpp#L365
```suggestion
    if (cudf::is_nested(output_dtype)) {
```